packages/fields/src/fields/use-post-fields/index.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useEntityRecords } from '@wordpress/core-data';
+import type { Field } from '@wordpress/dataviews';
+
+/**
+ * Internal dependencies
+ */
+import featuredImageField from '../featured-image';
+import slugField from '../slug';
+import parentField from '../parent';
+import passwordField from '../password';
+import statusField from '../status';
+import commentStatusField from '../comment-status';
+import titleField from '../title';
+import dateField from '../date';
+import authorField from '../author';
+import type { BasePost, BasePostWithEmbeddedAuthor } from '../../types';
+
+type Post = BasePost | BasePostWithEmbeddedAuthor;
+
+interface UsePostFieldsReturn {
+	isLoading: boolean;
+	fields: Field< Post >[];
+}
+
+interface Author {
+	id: number;
+	name: string;
+}
+
+function usePostFields(): UsePostFieldsReturn {
+	const { records: authors, isResolving: isLoadingAuthors } =
+		useEntityRecords< Author >( 'root', 'user', { per_page: -1 } );
+
+	const fields = useMemo(
+		() =>
+			[
+				featuredImageField,
+				titleField,
+				{
+					...authorField,
+					elements: authors?.map( ( { id, name } ) => ( {
+						value: id,
+						label: name,
+					} ) ),
+				},
+				statusField,
+				dateField,
+				slugField,
+				parentField,
+				commentStatusField,
+				passwordField,
+			] as Field< Post >[],
```

Comment:
Should this be `Field< Post >` or `Field< BasePostWithEmbeddedAuthor >`. My reading is the latter since the author field is always returned?

Reply:
Naming here may be confusing: note that I used `Post` but it's the one defined in this file above (so it considers both potential types):

```js
type Post = BasePost | BasePostWithEmbeddedAuthor;`
```

Happy to hear alternatives about how to better encapsulate this.

Reply:
It's not about naming, ` BasePost | BasePostWithEmbeddedAuthor` means usePostFields can be used for posts without authors, which is not really the case in the code because "author" is always returned.

Reply:
Ah, I see your point now. Fixed at https://github.com/WordPress/gutenberg/pull/67024/commits/d049eee629ffe118ed246c9c6183e83c1ae6354f